backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala

```diff
@@ -376,13 +376,8 @@ object VeloxBackendSettings extends BackendSettingsApi {
       } else {
         t match {
           // OPPRO-266: For Velox backend, build right and left are both supported for
-          // LeftOuter and LeftSemi.
-          // FIXME Hongze 22/12/06
-          //  HashJoin.scala in shim was not always loaded by class loader.
-          //  The file should be removed and we temporarily disable the improvement
-          //  introduced by OPPRO-266 by commenting out the following prerequisite
-          //  condition.
-//          case LeftOuter | LeftSemi => true
+          // LeftOuter.
+          case LeftOuter => true
```

Comment:
Build left is supported by LeftSemi on Velox backend. Shall we add it?

Reply:
The previous regression of b14a/b14b was due to left semi join build left side. Although build left is supported by LeftSemi on Velox backend, but the performance is not as good as build right, vanilla spark also choose build right.

Reply:
This function returns if building left is supported, which I assume indicates the backend's capability and shouldn't be mixed up the build side selection.

> vanilla spark also choose build right.

It is due to some special logic, e.g. preferredBuildSide, Gluten chooses to build left? If so, I think we need to fix there.

Reply:
I was planning to remove `preferredBuildSide` in the next PR because it is unnecessary and not as accurate as AQE. Like vanilla spark, its performance improvement is compared to sort merge join, for left outer and join outer, if both are shuffle hash joins, choosing a small table will not improve performance.
I don’t know the logic of velox’s support for small table build.  Shuffle hash join is forced to be used here, 
what I'm worried about is that for shuffle hash join of LeftSemi, build left is much slower than build right.

Reply:
https://github.com/apache/spark/blob/branch-3.2/sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q14a/explain.txt
Both left and right contain filter, so `isPreferred(left)` and `isPreferred(right)` both are true, so I think it is not caused `preferredBuildSide`.

Reply:
Not exactly. The two sides of the time-consuming join have a large difference on data size, so we need to make sure the smaller table is used as build side. I'm still confirming that.

![image](https://github.com/apache/incubator-gluten/assets/41687378/44181b00-ca81-4152-af63-796bd17af1c8)

Reply:
Could you add a TODO here to describe why we don't add LeftSemi? Maybe also include the issue link.
https://github.com/facebookincubator/velox/issues/9980

Reply:
Added.